addons/point_of_sale/static/src/js/models.js

```diff
@@ -909,29 +909,29 @@ exports.PosModel = Backbone.Model.extend({
             partnerModel.loaded(this, partners);
         }
     },
-    loadProductsFromOffset: async function(offset) {
-        let ProductIds = [];
+    // Load the products into the `db` and return the amount loaded
+    loadProductsFromPage: async function(page) {
         let product_model = _.find(this.models, (model) => model.model === 'product.product');
-        ProductIds = await this.rpc({
-            model: 'product.product',
-            method: 'search',
-            args: [product_model.domain(this)],
+        const products = await this.rpc({
+            model: 'pos.config',
+            method: 'get_limited_products_loading',
+            args: [[this.config_id]],
             kwargs: {
-                offset: this.env.pos.config.limited_products_amount * offset,
-                limit: this.env.pos.config.limited_products_amount,
+                fields: product_model.fields,
+                page,
             },
-            context: this.env.session.user_context,
+            context: { ...this.session.user_context, ...product_model.context() },
```

Comment:
I'm not sure if this is already allowed. If not, you can do `Object.assign(this.session.user_context, product_model.context())`.

Reply:
I'll just tag this as "Resolved" since the branch is green, not complaining about the use of `{...obj1, ...obj2}`.